src/dsm/headers/Dynamics.hpp

```diff
@@ -371,7 +371,7 @@ namespace dsm {
         }
       } else if (!agent->streetId().has_value()) {
         assert(agent->srcNodeId().has_value());
-        auto srcNode{this->m_graph->nodeSet()[agent->srcNodeId().value()]};
+        auto& srcNode{this->m_graph.nodeSet()[agent->srcNodeId().value()]};
```

Comment:
Why not `const auto&`?

Reply:
Because you go on to add an agent, which means that the node reference can't be const

Reply:
I'm not sure about it... maybe you are thinking about `std::unique_ptr<const Node>`?